drivers/ma_yya_report/app/models/ma_yya_report/universe_calculator.rb

```diff
@@ -0,0 +1,206 @@
+      enrollment = enrollments.last
+      enrollment.enrollment.ReferralSource.in?([1, 2, 11, 18, 28, 30, 34, 45, 37, 38, 39]) &&
+        enrollments[0...-1].detect { |en| en.first_date_in_program >= @filter.start_date - 24.months }.blank?
+    end
+
+    private def direct_assistance?(enrollment)
+      enrollment.
+        events.
+        select { |event| event.EventDate.between?(@filter.start_date, @filter.end_date) }.
+        any? { |event| event.Event == 16 }
+    end
+
+    private def gender(client)
+      genders = client.gender_multi
+      return 0 if genders == [0]
+      return 1 if genders == [1]
+      return 6 if genders.include?(6)
+      return 4 if genders.include?(4) || genders == [0, 1] || genders == [0, 1, 5]
+      return 5 if genders == [0, 5] || genders == [1, 5] || genders == [5] # rubocop:disable Style/MultipleComparison
+
+      return client.GenderNone
+    end
+
+    private def race(client)
+      return client.RaceNone if client.RaceNone.in?([8, 9, 99])
+
+      race_fields = client.race_fields
+      return 99 if race_fields.size.zero?
+      return 6 if race_fields.size > 1
+
+      return race_code[*race_fields]
+    end
+
+    private def race_code
+      {
+        'AmIndAKNative' => 1,
+        'Asian' => 2,
+        'BlackAfAmerican' => 3,
+        'NativeHIOtherPacific' => 4,
+        'White' => 5,
+      }.freeze
+    end
+
+    private def disability?(enrollment, disability_type, disability_responses = [1])
+      # Find most recent disability record associated with the enrollment for the appropriate type with a value
+      # recorded before the end of the reporting period
+      disability = enrollment.disabilities.detect do |d|
+        d.InformationDate < @filter.end_date &&
+          d.DisabilityType == disability_type &&
+          d.DisabilityResponse.in?([0, 1, 2, 3])
+      end
+      disability.present? && disability.IndefiniteAndImpairs == 1 && disability.DisabilityResponse.in?(disability_responses)
+    end
+
+    private def household_ages(enrollment)
+      enrollment.household_enrollments.map do |en|
+        next if en.EntryDate > @filter.end_date
+        next if en.exit.present? && en.exit.ExitDate < @filter.start_date
+
+        en.client.age_on([@filter.start_date, en.EntryDate].max)
+      end.compact
+    end
+
+    private def rehoused_on(enrollment)
+      enrollment.current_living_situations.
+        detect do |cls|
+          cls.CurrentLivingSituation.in?([19, 3, 31, 33, 34, 10, 20, 21, 11]) &&
+          cls.InformationDate > enrollment.EntryDate
+        end&.InformationDate
+    end
+
+    private def subsequent_current_living_situations(enrollment)
+      enrollment.current_living_situations.
+        select do |cls|
+        cls.InformationDate.between?(@filter.start_date, @filter.end_date) &&
+          cls.InformationDate >= enrollment.EntryDate + 90.days
+      end.
+        map(&:CurrentLivingSituation)
+    end
+
+    private def zip_code(client)
```

Comment:
I _think_ the client you are passing in here is a destination client, but the hmis_client record will be joined to source clients.

Reply:
Thanks! Good catch!